src/contrib/persistence/Akka.Persistence.Sql.Linq2Db/Journal/Linq2DbWriteJournal.cs

```diff
@@ -0,0 +1,189 @@
+        {
+            Context.System.Log.Error(cause,
+                $"Linq2Db Journal Error on {message?.GetType().ToString() ?? "null"}");
+            base.AroundPreRestart(cause, message);
+        }
+
+        
+        public override async Task ReplayMessagesAsync(IActorContext context, string persistenceId,
+            long fromSequenceNr, long toSequenceNr, long max, Action<IPersistentRepresentation> recoveryCallback)
+        {
+            await _journal.MessagesWithBatch(persistenceId, fromSequenceNr,
+                    toSequenceNr, _journalConfig.DaoConfig.ReplayBatchSize,
+                    Option<(TimeSpan, IScheduler)>.None)
+                .Take(max).SelectAsync(1,
+                    t => t.IsSuccess
+                        ? Task.FromResult(t.Success.Value)
+                        : Task.FromException<ReplayCompletion>(
+                            t.Failure.Value))
+                .RunForeach(r =>
+                {
+                    recoveryCallback(r.repr);
+                }, _mat);
+            
+        }
+
+        public override async Task<long> ReadHighestSequenceNrAsync(string persistenceId, long fromSequenceNr)
+        {
+            if (writeInProgress.ContainsKey(persistenceId))
+            {
+                try
+                {
+                    await writeInProgress[persistenceId];
+                }
+                catch (Exception)
+                {
+                    //We don't have 'Recover' in C# so this is intentionally empty
+                    //Basically we just wanted to wait for write to succeed OR fail
+                }
+                var hsn =await _journal.HighestSequenceNr(persistenceId,
+                    fromSequenceNr);
+                return hsn;
+            }
+            return await _journal.HighestSequenceNr(persistenceId, fromSequenceNr);
+        }
+        private Dictionary<string,Task> writeInProgress = new Dictionary<string, Task>();
+        
+        protected override async Task<IImmutableList<Exception>>
+            WriteMessagesAsync(IEnumerable<AtomicWrite> messages)
+        {
+            //TODO: CurrentTimeMillis;
+            var currentTime = DateTimeHelpers.UnixEpochMillis();
+            var persistenceId = messages.Head().PersistenceId;
+            var future = _journal.AsyncWriteMessages(messages,currentTime);
+            
+            writeInProgress.AddOrSet(persistenceId, future);
+            var self = Self;
+            
+            //When we are done, we want to send a 'WriteFinished' so that
+            //Sequence Number reads won't block/await/etc.
+            future.ContinueWith((p) =>
+                    self.Tell(new WriteFinished(persistenceId, future)),
+                CancellationToken.None,
+                TaskContinuationOptions.ExecuteSynchronously,
+                TaskScheduler.Default);
+            //But we still want to return the future from `AsyncWriteMessages`
+            return await future;
```

Comment:
May want to Double-Check Porting from [here](https://github.com/akka/akka-persistence-jdbc/blob/master/core/src/main/scala/akka/persistence/jdbc/journal/JdbcAsyncWriteJournal.scala#L94-L98)